src/scancode/cli.py

```diff
@@ -1176,10 +1176,11 @@ def run_scanners(
 
     progress_manager = None
     if not quiet:
+        resources = list(((r.location, r.path) for r in codebase.walk() if r.is_file))
```

Comment:
This will be a problem when we have a codebase with 10 million files, we may need to find a better way? ... otherwise the system will literally be stalled for minutes before displaying a progress bar.

Reply:
For the kernel, the delay is 45 sec until it traverses 200k files. While the scan itself runs several hours.
With -q there is no delay.